### DISCUSSION TITLE
消息发送前拦截

### DISCUSSION THREAD

**Discussion by Redmomn** [nonebot/discussions#375] [问答]
我有一个插件，发送的消息不可控，请问如何使用钩子函数在发送前检查违禁词并取消发送消息

**Comment by yanyongyu** [comment]
通过api调用前钩子，当调用发送消息的时候检测发送的消息内容，然后可以进行修改或阻止调用并报错或返回一个虚拟的结果

**Reply by Redmomn** [reply]
我知道是这个思路，但是要具体怎么写呢，求大佬给个示例代码

**Reply by yanyongyu** [reply]
https://v2.nonebot.dev/docs/next/advanced/runtime-hook#%E5%B9%B3%E5%8F%B0%E6%8E%A5%E5%8F%A3%E8%B0%83%E7%94%A8%E9%92%A9%E5%AD%90
